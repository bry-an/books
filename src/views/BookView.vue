<template>
  <div class="ma-2">
    <v-btn class="mb-3" @click="routeHome">Go back</v-btn>
    <div class="title">
      More information about
      <em>{{book.title}}</em>
    </div>
    <div class="subheading mt-3">{{book.title}} was written by {{book.author_name[0]}}</div>
    <div class="subheading mt-3">It was originally published in {{book.first_publish_year}}</div>
    <div class="subheading mt-3">Its ISBN is {{book.isbn[0]}}</div>
  </div>
</template>

<script>
export default {
  name: "BookView",
  data() {
    return {
      book: this.$route.params
    };
  },
  methods: {
    routeHome() {
      this.$router.push({ name: "Home" });
    }
  },
  mounted() {
    if (!this.$route.params) {
      this.$router.push({ name: "Home" });
    }
  }
};
</script>

<style lang="scss" scoped>
</style>