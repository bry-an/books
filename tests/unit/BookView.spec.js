
import { mount } from "@vue/test-utils"
import Vuetify from "vuetify"
import BookView from "../../src/views/BookView.vue"
import Vue from "vue"
Vue.use(Vuetify)

describe("Book View", () => {
    let build
    build = () => {

        const wrapper = mount(BookView, {
            mocks: {
                $route: {
                    params: {
                        title: "",
                        author_name: [""],
                        first_publish_year: 0,
                        isbn: [0]
                    }
                }
            }
        })
        wrapper.vm.$route.params = {
            title: "",
            author_name: [""],
            first_publish_year: 0,
            isbn: [0]
        }
        return {
            wrapper
        }
    }
    it("renders the page", () => {
        const { wrapper } = build()
        expect(wrapper.html()).toMatchSnapshot()
    })
    it("displays book information", () => {
        const { wrapper } = build()
        const bookTitle = wrapper.find(".title")
        expect(bookTitle.exists()).toBe(true)
    })
})
