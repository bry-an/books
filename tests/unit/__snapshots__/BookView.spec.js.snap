// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book View renders the page 1`] = `
<div class="ma-2"><button type="button" class="mb-3 v-btn theme--light">
    <div class="v-btn__content">Go back</div>
  </button>
  <div class="title">
    More information about
    <em></em></div>
  <div class="subheading mt-3"> was written by </div>
  <div class="subheading mt-3">It was originally published in 0</div>
  <div class="subheading mt-3">Its ISBN is 0</div>
</div>
`;
